### PR TITLE
Replace Danger / Slather with Codecov

### PR DESCRIPTION
**Pull request by withinfocus**
## 🎟️ Tracking

Internal change.

## 📔 Objective

Removes Danger and Slather in favor of our organization-wide coverage tool Codecov.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ env:
   MINT_LINK_PATH: .mint/bin
   MINT_PATH: .mint/lib
   SIMULATOR_DESTINATION: platform=iOS Simulator,name=iPhone 15 Pro,OS=17.0.1
+  RESULT_BUNDLE_PATH: build/AuthenticatorTests.xcresult
```
Comment (KatherineInCode) [inline .github/workflows/test.yml:17]:
```suggestion
  RESULT_BUNDLE_PATH: build/BitwardenTests.xcresult
```

**Review by KatherineInCode — CHANGES_REQUESTED** [review]
Looks good, just a small place where PM is different from BWA

File: .github/workflows/test.yml (hunk 1 above)
Comment (withinfocus) [reply .github/workflows/test.yml:17]: You're looking at drafts! I was mindlessly copying ... thanks for the catch.

**Comment by withinfocus** [comment]
Test workflow is expected to fail given the Danger removal.

**Review by fedemkr — COMMENTED** [review]
🤔 Would it be necessary to update the `README` to say **Codecov** is a third-party dependency to follow the previous approach?

File: .github/workflows/test.yml (hunk 1 above)
Comment (KatherineInCode) [reply .github/workflows/test.yml]: I completely missed that it was still a draft, sorry 😅

**Review by KatherineInCode — APPROVED** [review]

**Comment by withinfocus** [comment]
> 🤔 Would it be necessary to update the `README` to say **Codecov** is a third-party dependency / is being used in CI to follow the previous approach?

We don't note this anywhere.

**Pull request merged**